### PR TITLE
Add search box to document indexes

### PR DESCRIPTION
**Pull request by danielroseman**
Passes the q parameter on to publishing-api for filtering by title/slug.

### CODE REVIEW

File: app/models/document.rb [hunk 1]
```diff
@@ -182,9 +173,10 @@ def self.all(page, per_page)
         :publication_state,
       ],
       page: page,
-      per_page: per_page
-    ).to_ostruct
```
Comment (h-lame) [inline app/models/document.rb:186]: One less `OpenStruct` in the world! 🎏 🎆

File: spec/models/cma_case_spec.rb [hunk 2]
```diff
@@ -117,6 +117,13 @@ def cma_case_content_item(n)
     it "returns all CMA Cases" do
       expect(described_class.all(page, per_page).results.length).to be(cma_cases.length)
     end
+
+    it "passes query parameter if supplied" do
+      expect(described_class.publishing_api).to receive(:get_content_items)
+        .with(hash_including(q: "foo"))
+        .and_return(double(to_ostruct: {}))
+      described_class.all(page, per_page, q: "foo")
+    end
```
Comment (h-lame) [inline spec/models/cma_case_spec.rb:126]: Do we also want to have a feature spec here to make sure the UI -> model part works? (I'm sure it does, but...).

Comment (benilovj) [inline spec/models/cma_case_spec.rb:126]: :+1:

**Comment by danielroseman** [comment]
Added feature spec.

**Pull request merged**